src/services/storage/storage.service.ts

```diff
@@ -0,0 +1,183 @@
+import { Network } from '../../model/storage/network.model'
+import { Swarm } from '../../model/storage/swarm.model'
+import { removeAllValues } from '../../utils/array'
+import { networkFactory, networkListFactory, swarmFactory } from './storage-factories'
+import migrate from './storage-migration'
+
+/**
+ * Sets any value to the extension storage
+ * @param key entry's key
+ * @param value any value
+ * @returns promise
+ */
+export function setEntry<T>(key: string, value: T): Promise<void> {
+  return chrome.storage.local.set({ [key]: value })
+}
+
+/**
+ * Retreives value of provided key from the extension storage
+ * @param key entry's key
+ * @returns value from the extension storage
+ */
+export async function getEntry<T>(key: string): Promise<T> {
+  const data = await chrome.storage.local.get([key])
+
+  return data[key] as T
+}
+
+/**
+ * Deletes value from the extension storage
+ * @param key entry's key
+ * @returns promise
+ */
+export function deleteEntry(key: string): Promise<void> {
+  return chrome.storage.local.remove([key])
+}
+
+/**
+ * Stores an object to the extension storage.
+ * The new object is merged to the existing one. Only shallow marge is performed.
+ * @param key entry's key
+ * @param object object to store
+ * @returns promise
+ */
+export async function setObject<T extends object>(key: string, object: T): Promise<void> {
```

Comment:
I would rename it to something else as it merges already set values instead of overwriting

Reply:
Hmm, maybe `updateObject` has more sense.